### DISCUSSION TITLE
`docker run` not working inside a GitHub action

### DISCUSSION THREAD

**Discussion by JJ** [community/community#27209] [Actions]
<p>This line</p>
<pre><code class="lang-bash"> docker run -t -v `pwd`:/test ${CONTAINER,,} bash -c "ls &amp;&amp; ${IV_ORDEN_TAREAS} build &amp;&amp; ${IV_ORDEN_TAREAS} install &amp;&amp; ${IV_ORDEN_TAREAS} test"
</code></pre>
<p>inside a <a href="https://github.com/JJ/IV-20-21/runs/1525235660?check_suite_focus=true" rel="noopener nofollow ugc">Github action</a> fails with:</p>
<pre><code class="lang-nohighlight">docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "bash": executable file not found in $PATH: unknown.
</code></pre>
<p>It works without a problem locally from the command line. Anything you put as “bash” will not work either. I can’t use the same container as a runner, since its values are defined in some environment variables, so it’s not known at the time of starting the workflow.</p>
<p>Alternatively, being able to set some variables (as a matter of fact, they’re contained in an artifact) so that a container runner could be used from the beginning would be nice alternative.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Looking at the <a href="https://hub.docker.com/r/antoniocuadros/whentoclass/dockerfile">Dockerfile of that container</a> I’d be extremely surprised if Bash was available. It’s based on an Alpine image, which by default only has the Busybox shell (as <code>sh</code>), and there’s no command in the Dockerfile that looks like it would install Bash. Maybe you have an older version locally that included Bash?</p>
<p>Does your build work if you call <code>sh</code> instead of <code>bash</code>?</p>

**Comment by JJ** [comment]
<p>It actually does. Only it was not the first error I obtained, <a href="https://github.com/JJ/IV-20-21/runs/1525172815?check_suite_focus=true" rel="noopener nofollow ugc">ŧhis one for instance didn’t find <code>rake</code></a>, and others didn’t find “make” and other things that were actually installed. I eventually fixed this problem by using “sh -c”, as you have said, but I’d still like to know why something like docker run container make whatever works locally, but it does not work in a GitHub action.</p>

**Comment by airtower-luna** [comment]
<p>In the workflow you linked the problem was that the <code>docker exec</code> command didn’t call <code>rake</code>, it tried to call <code>"rake"</code> (including the quotes). :wink:</p>

**Comment by JJ** [comment]
<p>Ah, right. Stupid me. Should have eliminated the quotes from the variable. You’re amazing. Thanks!</p>
